packages/roosterjs-editor-dom/lib/list/VList.ts

```diff
@@ -207,8 +207,20 @@ export default class VList {
      * @param type Type of this list item, can be ListType.None
      */
     appendItem(node: Node, type: ListType) {
-        node = getTagOfNode(node) == 'LI' ? node : wrap(node, 'li');
-        this.items.push(type == ListType.None ? new VListItem(node) : new VListItem(node, type));
+        let nodeToAppend: Node;
+        if (getTagOfNode(node) != 'LI') {
+            nodeToAppend = document.createElement('LI');
+
+            while (node.hasChildNodes()) {
+                nodeToAppend.appendChild(node.firstChild);
+            }
+        } else {
+            nodeToAppend = node;
+        }
```

Comment:
Consider use util function "changeElementTag" https://github.com/microsoft/roosterjs/blob/d246c0c/packages/roosterjs-editor-dom/lib/utils/changeElementTag.ts#L10

Reply:
Great suggestion, I'll do that

Reply:
One more concern here. What if the original tag is a structured node, such as a TABLE, in that case we should not change its tag name. What do you think?

Reply:
Yea, I agree